File: neetcode/go/_121/main.go
package main

import "fmt"

func main() {
	fmt.Println(maxProfit([]int{7, 1, 5, 3, 6, 4}))
	fmt.Println(maxProfit([]int{7, 6, 4, 3, 1}))
}

func maxProfit(prices []int) int {
	l, r := 0, 1

	result := 0
	for r < len(prices) {
		if prices[l] < prices[r] {
			result = max(result, prices[r]-prices[l])
		} else {
			l = r
		}

		r++
	}

	return result
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: neetcode/python/_121/main.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        best = prices[0]
        res = 0
        for i in range(1, len(prices)):
            res = max(res, prices[i] - best)
            best = min(best, prices[i])
        return res

print(Solution().maxProfit([7,1,5,3,6,4]))